rados/read_op_exec.go

```diff
@@ -0,0 +1,114 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <stdlib.h>
+// #include <rados/librados.h>
+//
+import "C"
+
+import (
+	"runtime"
+	"unsafe"
+)
+
+// ReadOpExecStep - step for exec operation code.
+type ReadOpExecStep struct {
+	charPtr *C.char
+	size    C.size_t
+	prval   *C.int
+}
+
+// newExecStepOp - init new *execStepOp.
+func newReadOpExecStep() *ReadOpExecStep {
+	es := &ReadOpExecStep{
+		charPtr: nil,
+		size:    0,
+		prval:   (*C.int)(C.malloc(C.sizeof_int)),
+	}
+	*es.prval = 0
+
+	runtime.SetFinalizer(es, func(es *ReadOpExecStep) {
+		if es != nil {
+			es.free()
+			es.freeBuffer()
+			es = nil
+		}
+	})
+	return es
+}
+
+// free - will be called to free any resources, especially C memory.
+func (es *ReadOpExecStep) free() {
+	if es.prval != nil {
+		C.free(unsafe.Pointer(es.prval))
+		es.prval = nil
+	}
+}
+
+// freeBuffer - releases C allocated buffer. It is separated from es.free() because lifespan of C allocated buffer is
+// longer than lifespan of read operation.
+func (es *ReadOpExecStep) freeBuffer() {
+	if es.charPtr != nil {
+		C.free(unsafe.Pointer(es.charPtr))
+		es.charPtr = nil
+	}
+}
+
+// update - update state operation.
+func (es *ReadOpExecStep) update() error {
+	return getError(*es.prval)
+}
+
+// Bytes - go slice of bytes allocated in C.
```

Comment:
```suggestion
// Bytes returns the result of the executed command as a byte slice.
```
It's more idiomatic for Go to use a complete sentence in a doc comment. Also, the fact that it's allocated by C is an internal detail a user of the API should not need to worry about.  If you want to you can add a different (non-doc) comment to the function if you think it warrants a reminder that the memory was allocated by C.